src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -371,14 +348,45 @@ private void HandleSubscription(KafkaConsumerActorMetadata.Internal.ISubscriptio
         {
             try
             {
-                if (subscriptionRequest is KafkaConsumerActorMetadata.Internal.Subscribe subscribe)
-                    _consumer.Subscribe(subscribe.Topics);
-                else if (subscriptionRequest is KafkaConsumerActorMetadata.Internal.SubscribePattern subscribePattern)
-                    _consumer.Subscribe(subscribePattern.TopicPattern);
-                else
-                    throw new NotSupportedException($"Unsupported subscription type: {subscriptionRequest.GetType()}");
-                
+                switch (subscriptionRequest)
+                {
+                    case KafkaConsumerActorMetadata.Internal.Assign assign:
+                    {
+                        CheckOverlappingRequests("Assign", Sender, assign.TopicPartitions);
+                        var previousAssigned = _consumer.Assignment;
+                        _consumer.Assign(assign.TopicPartitions.Union(previousAssigned));
+                        _commitRefreshing.AssignedPositions(assign.TopicPartitions, _consumer, _settings.PositionTimeout);
+                        break;
+                    }
+
+                    case KafkaConsumerActorMetadata.Internal.AssignWithOffset assignWithOffset:
+                    {
+                        var topicPartitions = assignWithOffset.TopicPartitionOffsets.Select(o => o.TopicPartition).ToImmutableHashSet();
+                        CheckOverlappingRequests("AssignWithOffset", Sender, topicPartitions);
+                        
+                        // TODO: dear lord this is wrong, WE SHOULD NOT BE SETTING OFFSETS TO ZERO HERE
```

Comment:
TL;DR; not sure how bad of a bug this actually is but it is giving off code smell.